src/OrleansRuntime/GrainDirectory/ClientObserverRegistrar.cs

```diff
@@ -117,16 +119,8 @@ internal void ClientDropped(GrainId clientId)
             }
             catch (Exception exc)
             {
-                int actualExceptions = 1;
-                if (exc is AggregateException)
-                {
-                    AggregateException aggregateException = exc as AggregateException;
-                    actualExceptions = aggregateException.InnerExceptions.Count;
-                    exc = aggregateException.InnerExceptions.First();
-                }
                 logger.Error(ErrorCode.ClientRegistrarTimerFailed, 
-                    String.Format("OnClientRefreshTimer has thrown {0} inner exceptions. Printing the first exception:", actualExceptions), 
-                    exc);
+                    String.Format("OnClientRefreshTimer has thrown an exceptions."), exc);
```

Comment:
Hey gabi, following up on the gitter chat where you didn't just want the first exception, there's some alternatives.
In this case, since you are using `await Task.WhenAll`, you should be able to still see all the exceptions from all registration tasks. In the past you were catching and getting an AggregateException since Orleans was adding an extra wrapping on top of it, but still you were seeing ONLY one of the registration failures and not all, as thay is how await on Task.WhenAll works. When you use Task.WhenAll (regardless of Orleans) if you want to see all of the exceptions, you can explicitly store the Task.WhenAll task in a variable, and when an exception occurs, inspect the Exception property.
Nevertheless, in this case I don't see a reason to do that, since you were printing the 1st exception anyway AND each registration failure is already logging the the exception with the LogException extension method.
What I wanted to say is that I think this is now correct regardless of the changes to the wrapping that we are no longer doing, but if I'm missing something, let me know and I can help you get right semantics.